api/lib/db-interface.js

```diff
@@ -226,6 +243,10 @@ DI.saveTransaction = function(transaction, callback) {
                 + err);
 
     (callback || noop)(err);
+  })
+  .finally(function() {
+    self.lock[txData.client_resource_id] = undefined;
```

Comment:
My recollection is that `delete` may trigger deoptimizations and should be avoided if at all possible. I'm more concerned about this